src/main/java/com/mysticalchemy/api/events/CrucibleEvent.java

```diff
@@ -0,0 +1,92 @@
+package com.mysticalchemy.api.events;
+
+import com.mysticalchemy.recipe.PotionIngredientRecipe;
+import net.minecraft.world.effect.MobEffect;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+import net.minecraftforge.eventbus.api.Cancelable;
+import net.minecraftforge.eventbus.api.Event;
+
+import java.util.HashMap;
+
+/**
+ * This event is fired when the crucible is used.
+ */
+public class CrucibleEvent extends Event {
+
+    public CrucibleEvent() {}
+
+
+    /**
+     * CrucibleEvent.AddIngredient is fired when an ingredient is added to the crucible.<br>
+     * <br>
+     * {@link #effects} contains the effects already stored in the crucible.<br>
+     * {@link #recipe} contains the {@code potion_ingredient recipe} that is being added to the crucible.<br>
+     * {@link #stack} contains the ItemStack that is being added to the crucible.<br>
+     * <br>
+     * The recipe can be modified during the event's execution to affect what happens when the ingredient is added.<br>
+     * The stack size can also be increased or decreased during the event's execution.<br>
```

Comment:
I question the decision here that stack size can be modified during the event's execution.  Can you remind me why this is needed?  It's generally discouraged as the event is more about the effects than the ingredient, this seems like a strange byproduct.

Reply:
The example I thought of for this was some charm or enchantment which, if present on the user, doubles input.  For example, a Witch (MNA Witchcraft) wearing their Witchy Armor, might be able to throw in one feather, and get the benefit of throwing in two feathers.

Because of the way canMerge and mergeEffects are written, if I can't modify the recipe (which I accept), then there is no way for me to adjust the effects _to be applied_ without manipulating the stack itself.  I can only modify the effects on the object itself, which ideally we don't do during the execution because some later Event handler may DENY the result and my Event handler won't even know about it.

I am very open to suggestions on alternative implementations, this was just my thought process at the time. If we _don't_ want to allow this, I will change EventDispatcher#16 to use a copy of the stack when it creates the event, to prevent modifying the underlying stack.

But that brings into question what this event is even _for_ anymore, since now you can't modify the Recipe, and you can't modify the StackSize, and you _shouldn't_ modify the effects at this stage, so perhaps this event just gets removed or its results are largely ignored in favor of the ExtractPotion event.